src/main/java/io/orangebeard/listener/OrangebeardExtension.java

```diff
@@ -72,20 +76,68 @@ public OrangebeardExtension() {
         startTestRunAndAddShutdownHook(orangebeardProperties);
     }
 
+    /**
+     * Given an ExtensionContext, determine the fully qualified name (canonical name) of the class that it refers to.
+     * If the ExtensionContext does not have a class (as can happen in a unit test), then it returns an empty String.
+     * @param extensionContext ExtensionContext for a class that is being tested.
+     * @return The fully qualified name of the class that is being tested, or an empty String if there is no such class.
+     */
+    private String getFullyQualifiedClassName(ExtensionContext extensionContext) {
+        Class<?> requiredTestClass = extensionContext.getRequiredTestClass();
+        String fullyQualifiedName = "";
+        if (requiredTestClass != null) {
+            fullyQualifiedName = requiredTestClass.getCanonicalName();
+        }
+        return fullyQualifiedName;
+    }
+
     @Override
     public void beforeAll(ExtensionContext extensionContext) {
-        StartTestItem testSuite = new StartTestItem(testrunUUID, extensionContext.getDisplayName(), SUITE, null, null);
-        UUID suiteId = orangebeardClient.startTestItem(null, testSuite);
-        suites.put(extensionContext.getUniqueId(), new Suite(suiteId));
+
+        String fullyQualifiedName = getFullyQualifiedClassName(extensionContext);
+        String[] classNameComponents = fullyQualifiedName.split("\\.");
```

Comment:
I think the getFullyQualifiedClassName should handle everything with regard to splitting up the name and returning an array. So it should include the regex split.